ci/docker/Dockerfile

```diff
@@ -1,12 +1,13 @@
-FROM ubuntu:20.04
+FROM ubuntu:22.04
```

Comment:
oh, actually, it will only affect the tests that use Rust 1.80.1. This means v2.0 and v1.18 won’t be affected by these changes (and old master commits that are using Rust 1.79.0). do you still have concerns if the situation is as I described? to me, it's a really good chance to upgrade those things when we introduce a new rust version haha. just like a new start!

btw, I’m still struggling with the coverage tests 😢 I will keep posting here

Reply:
okay! let me try! I think I figured out the coverage issue. need to rebuild the image. will push commits and ping you!

Reply:
@yihau hmm, maybe i should have been more explicit at my suggestion of this: https://github.com/anza-xyz/agave/pull/2487#issuecomment-2287546356

i think we should use vendorized llvm from the rust-lang, instead of installing the official llvm manually.. this is easier and natually is sync with rust nightly more. in that way, bumping ubuntu (is good thing!) can be separated from this pr.

recent nightly's coverage data is incompatible with llvm 18. it only works now with llvm 19.

```
$ rustup component add --toolchain nightly-2024-08-08-x86_64-unknown-linux-gnu llvm-tools-preview
info: downloading component 'llvm-tools'
info: installing component 'llvm-tools'
 32.7 MiB /  32.7 MiB (100 %)  14.7 MiB/s in  2s ETA:  0s
$ rustup component add --toolchain 1.80.1-x86_64-unknown-linux-gnu llvm-tools-preview
info: downloading component 'llvm-tools'
info: installing component 'llvm-tools'
 31.2 MiB /  31.2 MiB (100 %)  14.8 MiB/s in  2s ETA:  0s

$ /home/ryoqun/.rustup/toolchains/1.80.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-profdata --version
LLVM (http://llvm.org/):
  LLVM version 18.1.7-rust-1.80.1-stable
  Optimized build.
$ /home/ryoqun/.rustup/toolchains/nightly-2024-08-08-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-profdata --version
LLVM (http://llvm.org/):
  LLVM version 19.1.0-rust-1.82.0-nightly
  Optimized build.

```

After that, I think we need to just meddle with PATH only running the `grcov`.

Reply:
(nevertheless, thanks for working on bumping rust. as i like to use latest and greatest rust things (yet not enough time...), i'd like to appreciate your efforts very much.)